lib/mail/matchers/has_sent_mail.rb

```diff
@@ -51,6 +51,10 @@ def bcc(recipient_or_list)
         self
       end
 
+      def with_attachments(*attachments)
+        @attachments = attachments
+        self
+      end
```

Comment:
Ah, I didn't previously understand that you were pointing towards [matching arguments](https://relishapp.com/rspec/rspec-mocks/v/3-3/docs/setting-constraints/matching-arguments). If I were to impement `any_attachment`, or `an_attachment_with_filename(filename)`, what do you envision chaining them them together would look like? or would there never be intent to create a test like
` it { is_expected_to have_sent_email.with_attachments(any_attachment).with_attachments(an_attachment_with_filename('foo')) }`